pandas/_testing/asserters.py

```diff
@@ -1382,3 +1382,20 @@ def assert_metadata_equivalent(
             assert val is None
         else:
             assert val == getattr(right, attr, None)
+
+
+def assert_mixed_int_string_entry(arr) -> bool:
+    """
+    Check that arr is mixed-int-string entry.
```

Comment:
I think this will be clearer if it refers directly to the relevant fixture.

Reply:
i think this is an instance of a more general pain point in test writing:

```
def test_something(fixture):
    if the_fixture_value_is_this_particular_entry(fixture):
        ...
```

usually we either inspect the object like this assert_mixed_int_string_entry is doing or in some cases we pass `request` and do something hacky with request.fixturenames or something.  I wonder if there is a recommended pattern for like `pytest.get_active_fixture_id("index")` that would return "mixed-int-string" here

Reply:
I coud't find a recommended pattern that would return `"mixed-int-string"` in this case. 
Instead I passed `request` and used `request.node.callspec.id` to return `"mixed-int-string"`.

Now I inspect the object by using `request.node.callspec.id` instead of  `is_mixed_int_string_entry`. The function `is_mixed_int_string_entry` is removed.
Sorry it took so long. Could you please take a look at this PR?